cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/table/StructuredTableSpecification.java

```diff
@@ -169,6 +202,17 @@ public Builder withFields(FieldType ...fieldTypes) {
       return this;
     }
 
+    /**
+     * Set the needToAddFields boolean for the table schema.
+     * This is optional and only for the purpose of adding new non-primary key columns.
+     * @param addFields boolean to indicate if it's needed to add new columns.
+     * @return Builder instance
+     */
+    public Builder withNeedToAddFields(boolean addFields) {
```

Comment:
should the param also be named as `needToAddFields`?

Reply:
updated